README.md

```diff
@@ -23,157 +24,226 @@ Following hardware was reported to work (with some limitations or workarounds)
-gpio4.on("change", processPin4);
-            
+gpioPin.on("change", processPin4);
+
 // unbind a particular callback from the "change" event
-gpio4.removeListener("change", processPin4);
-      
+gpioPin.removeListener("change", processPin4);
+
 // unbind all callbacks from the "change" event
-gpio4.removeAllListeners("change");
-      
-// you can also manually change the direction anytime after instantiation            
-gpio4.setDirection(gpio.DIRECTION.OUT);
-gpio4.setDirection(gpio.DIRECTION.IN);
+gpioPin.removeAllListeners("change");
 ```
 
-## Example
 
-### Cycle voltage every half a second
+## Basic Example
+
+Cycle voltage every half a second
 
 ```js
 var gpio = require("gpio");
-var gpio22, gpio4, intervalTimer;
 
 // Flashing lights if LED connected to GPIO22
-gpio22 = gpio.export(22, {
-   ready: function() {
-      intervalTimer = setInterval(function() {
-         gpio22.set();
-         setTimeout(function() { gpio22.reset(); }, 500);
-      }, 1000);
-   }
+var gpio22 = gpio.open({
+    pin: 22,
+    direction: gpio.DIRECTION.OUT
+}, function(err, pin) {
+    if(err) return console.log(err);
+
+    // blink at one second interval
+    var intervalTimer = setInterval(function() {
+        pin.set();
+        setTimeout(function() { pin.reset(); }, 500);
+    }, 1000);
+
+    // stop voltage cycling after 10 seconds
+    setTimeout(function() {
+        clearInterval(intervalTimer);
+    }, 10000);
 });
+```
+
+Expanding on above code, assume a different LED is hooked up to GPIO4.
+The following code will blink GPIO4 LED inversely with GPIO22 LED.
 
-// Lets assume a different LED is hooked up to pin 4, the following code 
-// will make that LED blink inversely with LED from pin 22 
-gpio4 = gpio.export(4, {
-   ready: function() {
-      // bind to gpio22's change event
-      gpio22.on("change", function(val) {
-         gpio4.set(1 - val); // set gpio4 to the opposite value
-      });
-   }
+```js
+var gpio4 = gpio.open({
+    pin: 4,
+    direction: gpio.DIRECTION.OUT
+}, function(err, pin) {
+    if(err) return console.log(err);
+
+    // bind to gpio22's change event
+    gpio22.on("change", function(val) {
+        // set gpio4 to the opposite value
+        pin.set(!val);
+    });
 });
+```
+
+Remember to close and cleanup your program when done
 
-// reset the headers and unexport after 10 seconds
+```js
 setTimeout(function() {
-   clearInterval(intervalTimer);          // stops the voltage cycling
-   gpio22.removeAllListeners('change');   // unbinds change event
-   gpio22.reset();                        // sets header to low
-   gpio22.unexport();                     // unexport the header
-   
-   gpio4.reset();
-   gpio4.unexport(function() {
-      // unexport takes a callback which gets fired as soon as unexporting is done
-      process.exit(); // exits your node program
-   });
+    gpio22.removeAllListeners('change');   // unbinds change event
+    gpio22.reset();                        // sets pin to low
```

Comment:
Technically setting/resetting isn't necessary as part of the cleanup process, probably better if I remove this reset call in the example altogether.

Re: deprecate set/reset - great idea, conforms to existing API standards. I might do this in a different PR since it touches a lot of files.